OpenRA.Mods.Common/Traits/World/SmudgeLayer.cs

```diff
@@ -163,14 +163,32 @@ public void AddSmudge(CPos loc)
 			}
 		}
 
+		public void RemoveSmudge(CPos loc)
+		{
+			if (dirty.ContainsKey(loc))
+			{
+				var tile = dirty[loc];
+				tile.Sprite = null;
+				dirty[loc] = tile;
+			}
+			else
+			{
+				var st = smudges.Keys.Random(world.SharedRandom);
+				dirty[loc] = new Smudge { Type = st, Depth = 0, Sprite = null };
```

Comment:
In that case, if a previous smudge exists, should we not blanket overwrite it? (i.e. reset the depth and randomize the type again) otherwise adding a new one will reuse the same type at whatever the old depth was, right?